tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -1398,12 +1398,18 @@ spv::LoopControlMask SpirvEmitter::translateLoopAttribute(const Stmt *stmt,
 
 void SpirvEmitter::doDiscardStmt(const DiscardStmt *discardStmt) {
   assert(!spvBuilder.isCurrentBasicBlockTerminated());
-  spvBuilder.createKill(discardStmt->getLoc());
-  // Some statements that alter the control flow (break, continue, return, and
-  // discard), require creation of a new basic block to hold any statement that
-  // may follow them.
-  auto *newBB = spvBuilder.createBasicBlock();
-  spvBuilder.setInsertPoint(newBB);
+
+  // The discard statement can only be called from a pixel shader
+  assert(spvContext.isPS());
```

Comment:
I am not sure what is the expected behavior from the AST parser, but I think we should print an explicit error message saying "The discard statement can only be called from a pixel shader".

I am just worried what if a user use release build and is not able to recognize it.

Reply:
Good idea. Done.